### PR TITLE
Use async/await

### PR DESCRIPTION
**Pull request by alalamav**
- Replaces promise chains with async/await syntax.
- Catches previously uncaught promise rejections and stops throwing in `main`.

### CODE REVIEW

File: src/www/app/electron_outline_tunnel.ts [hunk 1]
```diff
@@ -44,31 +44,33 @@ export class ElectronOutlineTunnel implements cordova.plugins.outline.Tunnel {
...
+      return;
     }
 
-    return promiseIpc.send('stop-proxying').then(() => {
+    try {
+      await promiseIpc.send('stop-proxying');
       this.running = false;
-    });
+    } catch (e) {
+      console.error(`Failed to stop tunnelL ${e}`);
```
Comment (JonathanDCohen) [inline src/www/app/electron_outline_tunnel.ts:68]: nit: capital L there at the end

**Review by JonathanDCohen — APPROVED** [review]

File: src/www/app/electron_outline_tunnel.ts (hunk 1 above)
Comment (alalamav) [reply src/www/app/electron_outline_tunnel.ts]: Done in ba6575b.

**Comment by alalamav** [comment]
Thanks for the review. I realized I forgot to convert `FakeOutlineTunnel` to async/await syntax, so I took the opportunity to do it now and rename the file (s/connection/tunnel).

File: src/www/app/fake_tunnel.ts [hunk 2]
```diff
@@ -28,40 +28,36 @@ export class FakeOutlineTunnel implements cordova.plugins.outline.Tunnel {
   }
 
   private playUnreachable() {
-    return !(this.config.name && this.config.name.toLowerCase().includes('unreachable'));
+    return this.config.name && this.config.name.toLowerCase().includes('unreachable');
```
Comment (JonathanDCohen) [inline src/www/app/fake_tunnel.ts:31]: while we're here, `this.config.name?.toLowerCase()...`

Comment (alalamav) [reply src/www/app/fake_tunnel.ts]: Done in 3e74749. Updated typescript to v3.7, which introduced optional support. I'll keep an eye out to use optionals where possible.

**Review by JonathanDCohen — APPROVED** [review]

**Pull request merged**